rust-runtime/aws-smithy-runtime/src/client/identity/cache/lazy.rs

```diff
@@ -0,0 +1,671 @@
+    ///
+    /// For example, if the identity is expiring in 15 minutes, the buffer time is 10 seconds,
+    /// and buffer time jitter fraction is 0.2, then buffer time is adjusted to 8 seconds.
+    /// Therefore, any requests made after 14 minutes and 52 seconds will load a new identity.
+    ///
+    /// Defaults to a randomly generated value between 0.0 and 1.0. This setter is for testing only.
+    #[cfg(feature = "test-util")]
+    pub fn buffer_time_jitter_fraction(mut self, buffer_time_jitter_fraction: fn() -> f64) -> Self {
+        self.set_buffer_time_jitter_fraction(Some(buffer_time_jitter_fraction));
+        self
+    }
+
+    /// A random percentage by which buffer time is jittered for randomization.
+    ///
+    /// For example, if the identity is expiring in 15 minutes, the buffer time is 10 seconds,
+    /// and buffer time jitter fraction is 0.2, then buffer time is adjusted to 8 seconds.
+    /// Therefore, any requests made after 14 minutes and 52 seconds will load a new identity.
+    ///
+    /// Defaults to a randomly generated value between 0.0 and 1.0. This setter is for testing only.
+    #[cfg(feature = "test-util")]
+    pub fn set_buffer_time_jitter_fraction(
+        &mut self,
+        buffer_time_jitter_fraction: Option<fn() -> f64>,
+    ) -> &mut Self {
+        self.buffer_time_jitter_fraction = buffer_time_jitter_fraction;
+        self
+    }
+
+    /// Default expiration time to set on an identity if it doesn't have an expiration time.
+    ///
+    /// This is only used if the resolved identity doesn't have an expiration time set.
+    /// This must be at least 15 minutes.
+    ///
+    /// Defaults to 15 minutes.
+    pub fn default_expiration(mut self, duration: Duration) -> Self {
+        self.set_default_expiration(Some(duration));
+        self
+    }
+
+    /// Default expiration time to set on an identity if it doesn't have an expiration time.
+    ///
+    /// This is only used if the resolved identity doesn't have an expiration time set.
+    /// This must be at least 15 minutes.
+    ///
+    /// Defaults to 15 minutes.
+    pub fn set_default_expiration(&mut self, duration: Option<Duration>) -> &mut Self {
+        self.default_expiration = duration;
+        self
+    }
+
+    /// Builds a [`SharedIdentityCache`] from this builder.
+    ///
+    /// # Panics
+    ///
+    /// This builder will panic if required fields are not given, or if given values are not valid.
+    pub fn build(self) -> SharedIdentityCache {
+        let default_expiration = self.default_expiration.unwrap_or(DEFAULT_EXPIRATION);
+        assert!(
+            default_expiration >= DEFAULT_EXPIRATION,
+            "default_expiration must be at least 15 minutes"
+        );
+        LazyCache::new(
+            self.time_source.unwrap_or_default(),
+            self.sleep_impl.unwrap_or_else(|| {
+                default_async_sleep().expect("no default sleep implementation available")
```

Comment:
should this come from a runtime plugin / runtime components?

Reply:
Good call out. I refactored it so that these come in through a `RuntimeComponents` argument on `ResolveIdentity` so that they will work correctly with config overrides.